src/nlsolve_ac_powerflow.jl

```diff
@@ -30,29 +30,33 @@ solve_ac_powerflow!(sys)
 solve_ac_powerflow!(sys, method=:newton)
 ```
 """
-function solve_ac_powerflow!(system::PSY.System; kwargs...)
+function solve_ac_powerflow!(
+    pf::Union{NLSolveACPowerFlow, KLUACPowerFlow},
+    system::PSY.System;
+    kwargs...,
+)
     #Save per-unit flag
     settings_unit_cache = deepcopy(system.units_settings.unit_system)
     #Work in System per unit
     PSY.set_units_base_system!(system, "SYSTEM_BASE")
     check_reactive_power_limits = get(kwargs, :check_reactive_power_limits, false)
     data = PowerFlowData(
-        ACPowerFlow(; check_reactive_power_limits = check_reactive_power_limits),
+        NLSolveACPowerFlow(; check_reactive_power_limits = check_reactive_power_limits),
         system;
         check_connectivity = get(kwargs, :check_connectivity, true),
     )
     max_iterations = DEFAULT_MAX_REDISTRIBUTION_ITERATIONS
-    res = _solve_powerflow!(data, check_reactive_power_limits; kwargs...)
-    if res.f_converged
-        write_powerflow_solution!(system, res.zero, max_iterations)
+    converged, x = _solve_powerflow!(pf, data, check_reactive_power_limits; kwargs...)
+    if converged
+        write_powerflow_solution!(system, x, max_iterations)
         @info("PowerFlow solve converged, the results have been stored in the system")
         #Restore original per unit base
         PSY.set_units_base_system!(system, settings_unit_cache)
```

Comment:
No need to do this now but just flagging that this is a good use case for https://github.com/NREL-Sienna/PowerFlows.jl/blob/465e92fcf921c460f1bd9e0c042c2cc335897c2f/src/psse_export.jl#L47 (and a good argument for moving it someplace more central)